### ISSUE TITLE
List Item Component: Nesting Enhancement and Drag and Drop

### ISSUE THREAD

**Issue by saumeel12** [Esri/calcite-design-system#3551]
### Description

Request for nesting functionality for the list item component. This would also work with the drag and drop functionality of the List item component.

### Acceptance Criteria 

The user should be able to nest list items under other list items either through a button or drag and drop.

### Relevant Info 

This enhancement was discussed with Alan and Mitch to determine this as an effective enhancement. This is planned tentatively for 10.1 release. This design also required Inline editing for list items, will tag that issue here shortly

#### Which Component

List Item

#### Example Use Case

This is useful for the Content/ Member categories use cases: Here the user starts off with one category (Which is a list item). The following three image will describe the creation and nesting process.

<img width="838" alt="Screen Shot 2021-11-22 at 2 06 19 PM" src="https://user-images.githubusercontent.com/87037561/142942155-d4e86cc7-614b-46ed-bf6c-cf1384ea5802.png">

In this design, the user can use the three dot menu to add subcategories under the main category; basically nesting list items under list items.

<img width="682" alt="Screen Shot 2021-11-22 at 2 12 01 PM" src="https://user-images.githubusercontent.com/87037561/142942771-b227fa6d-063a-4b13-8f2a-c82b6a4fcbd4.png">

This design restricts nesting up to two levels only.

<img width="701" alt="Screen Shot 2021-11-22 at 1 56 27 PM" src="https://user-images.githubusercontent.com/87037561/142941002-a72b0a56-824c-4fee-ac5b-91c5d47be00a.png">

The hierarchy can also be changed by dragging and dropping list items. See examples below:

<img width="638" alt="Screen Shot 2021-11-22 at 2 44 07 PM" src="https://user-images.githubusercontent.com/87037561/142946096-7ccdf752-4480-4889-8e98-3730b556ef2c.png">

Using the drag handle, the user can rearrange categories (List items). When dragging a list item and hovering between other list items; the list items should move down about 32 pixels and the black line should appear to indicate where the dragged list item will be placed.

<img width="500" alt="Screen Shot 2021-11-22 at 2 44 33 PM" src="https://user-images.githubusercontent.com/87037561/142946136-ccb5021f-321e-4cc8-a861-1015eb25b405.png">

If the user were to indent the dragged list item (Just by moving the mouse a bit to the right), the line length should indent as well, placing the list item down would nest the item under another category

<img width="502" alt="Screen Shot 2021-11-22 at 2 44 56 PM" src="https://user-images.githubusercontent.com/87037561/142946176-8e7355e8-dcb8-49e0-8c35-21a2b4831485.png">

Since in this project, we restrict nesting to two levels, we cannot allow the user to drag and nest in the following cases:

1) If the dragged list item has two nested list items and it's about to be nested under a single list item
2) If the dragged list item has one nested list item and its about to be nested under a first level nested list item
3) If the dragged list item is about to be nested under a second level nested list item

Basically, there should be no scenario where there are more than two levels of nesting

@asangma cc

**Comment by jcfranco** [comment]
Aside from the "add list item button", I think this should be possible with the latest drag-and-drop enhancements (including nesting) to `list` along with available `list-item` slots. 

Currently, we have no plans to add a button to the list for adding items, so this would have to be handled by the app or component using `list`.

cc @driskull

**Comment by driskull** [comment]
Yes, everything else should be possible now.

**Comment by jcfranco** [comment]
Closing based on the comments above.
